tslib/src/schemas/check-schemas/submitted-check.ts

```diff
@@ -0,0 +1,152 @@
+export interface SubmittedCheck {
+  checkCode: string
+  schoolUUID: string
+  config: CheckConfig
+  device: {
+    battery: {
+      isCharging: boolean
+      levelPercent: number
+      chargingTime: number
+      dischargingTime: number | null
+    }
+    cpu: {
+      hardwareConcurrency: number
+    }
+    navigator: {
+      userAgent: string
+      platform: string
+      language: string
+      cookieEnabled: boolean
+      doNotTrack: boolean | null
+    }
+    networkConnection: {
+      downlink: number
+      effectiveType: string
+      rtt: number
+    }
+    screen: {
+      screenWidth: number
+      screenHeight: number
+      outerWidth: number
+      outerHeight: number
+      innerWidth: number
+      innerHeight: number
+      colorDepth: number
+      orientation: string // TODO range of values?
+    }
+    deviceId: string
+    appUsageCounter: number
+  }
+  pupil: {
+    checkCode: string
+    inputAssistant: {
+      firstName: string
+      lastName: string
+    }
+  }
+  questions: CheckQuestion[]
+  school: {
+    name: string
+    uuid: string
+  }
+  tokens: {
+    checkStarted: QueueAuthToken
+    pupilPreferences: QueueAuthToken
+    pupilFeedback: QueueAuthToken
+    checkComplete?: QueueAuthToken
+    jwt: {
+      token: string
+    }
+  }
+  audit: CompleteCheckAuditEntry[]
+  inputs: CompleteCheckInputEntry[]
+  answers: CompleteCheckAnswer[]
+}
+
+export interface CheckConfig {
+  audibleSounds: boolean
+  checkTime: number
+  colourContrast: boolean
+  fontSize: boolean
+  inputAssistance: boolean
+  loadingTime: number
+  nextBetweenQuestions: boolean
+  numpadRemoval: boolean
+  practice: boolean
+  questionReader: boolean
+  questionTime: number
+  compressCompletedCheck: boolean
+}
+
+export interface QueueAuthToken {
+  token: string
+  url: string
+  queueName: string
+}
+
+export interface CheckQuestion {
+  order: number
+  factor1: number
+  factor2: number
+}
+
+export interface CompleteCheckAnswer {
+  factor1: number
+  factor2: number
+  answer: number
+  sequenceNumber: number
+  question: string
+  clientTimestamp: string
+}
+
+export enum InputEventType {
+  Touch = 'touch',
+  KeyDown = 'keydown',
+  Enter = 'enter'
+}
+
+export interface CompleteCheckInputEntry {
+  input: string
+  eventType: InputEventType
+  clientTimestamp: string
+  question: string
+  sequenceNumber: number
+  relativeTiming: string
```

Comment:
Please delete - it's not actually a part of the message.  The check viewer was lying.  Sorry.